mapper/block.go

```diff
@@ -17,11 +18,14 @@ func BlockMetadata(block *types.Block) map[string]interface{} {
 	if block.BaseFee() != nil {
 		meta["base_fee"] = hexutil.EncodeBig(block.BaseFee())
 	}
-	if block.BlockGasCost() != nil {
-		meta["block_gas_cost"] = hexutil.EncodeBig(block.BlockGasCost())
+
```

Comment:
You need to be using [emulation](https://github.com/ava-labs/avalanchego/pull/4430) here otherwise the `customtypes` package functions will panic.

That PR should be merged very soon, but in the meantime just look inside it to see the single function call made by `emulate.CChain()`.

If you're already wrapping at a higher level then it's ok and MUST NOT be done twice as nested calls to `WithTempRegisteredLibEVMExtras()` will deadlock as they attempt to hold the same `sync.Mutex`.

Reply:
Using `emulate.CChain` now 👍